src/pages/correspondent-agent-branches/Windows/AgentBranchWindow.js

```diff
@@ -1,73 +1,71 @@
-// ** Custom Imports
-import Window from 'src/components/Shared/Window'
 import CustomTabPanel from 'src/components/Shared/CustomTabPanel'
-import AgentBranchTab from 'src/pages/correspondent-agent-branches/Tabs/AgentBranchTab'
-import AddressTab from 'src/components/Shared/AddressTab'
+import AgentBranchForm from 'src/pages/correspondent-agent-branches/forms/AgentBranchForm'
+import AddressForm from 'src/components/Shared/AddressForm'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { CustomTabs } from 'src/components/Shared/CustomTabs'
+import { useState } from 'react'
+import { useContext } from 'react'
+import { RemittanceSettingsRepository } from 'src/repositories/RemittanceRepository'
+import toast from 'react-hot-toast'
 
-const AgentBranchWindow = ({
-  onClose,
-  width,
+const AgentBranchWindow = ({ labels, maxAccess, recordId, height }) => {
+  const [store, setStore] = useState({
+    recordId: recordId || null,
+    agentBranch: null,
+    address: null,
+    addressId: null
+  })
+
+  const [activeTab, setActiveTab] = useState(0)
+  const tabs = [{ label: labels.agentBranch }, { label: labels.address, disabled: !store.recordId }]
+  const { postRequest } = useContext(RequestsContext)
+
+  async function onSubmit(address) {
+     if (!store.agentBranch.addressId) {
+      store.agentBranch.addressId = address.addressId
+      const res = { ...store.agentBranch, addressId: address.addressId }
+      if (res) {
+        const data = { ...res, recordId: store?.recordId, agentId: store.agentBranch?.agentId }
+        await postRequest({
+          extension: RemittanceSettingsRepository.CorrespondentAgentBranches.set,
+          record: JSON.stringify(data)
+        })
+      }
+    } 
+  }
+
+  function setAddress(res) {
```

Comment:
Are you sure we need this? Can't we set the address from the onSubmit?
`setAddress` is legacy and I'm trying to limit usage of it.

Reply:
this is based on address form shared component concept